### DISCUSSION TITLE
737

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#918] [Comments]
# 737

Hibernate (Spring Data JPA) 를 사용하다보면 종종 HHH000104: firstResult/maxResults specified with collection fetch; applying in memory! 의 WARN (경고) 로그 메세지를 만난다. 해당 로그는 페이징 처리할때 여러 엔티티를 Fetch Join 을 하면 발생한다. Fetch Join은 N+1 문제를 해결하는 가장 자주 사용되던 방식이다. 하지만, 경고 메시지에서 언급했듯이 페이징 처리시에 사용할 경우 페이징이 전혀 적용되지 않고, 조건에 해당하는 모든 데이터를 가져와 메모리에 올려두고 사용한다. 조건에 해당 하는 데이터 전체를 가져오기 때문에 당연히 성능 상 이슈가 되며, 이를 메모리에 올려두고 페이징을 처리하니 이..

https://jojoldu.tistory.com/737

**Comment by Sirius506775** [comment]
감사히 잘 읽고 갑니다!!

**Comment by vact19** [comment]
감사감사 추수감사
